source/app/service-providers/fsal/util/search-file.ts

```diff
@@ -163,7 +167,9 @@ export default function searchFile (fileObject: MDFileDescriptor|CodeFileDescrip
   // Now immediately check if all required terms have matched. If not, we can
   // disregard this file now and save some computing time below.
   if (termsMatched !== termsToSearch.length) {
-    return []
+    // Make sure we return the finalResults array instead of an empty array, so
+    // we don't lose the file in case only its title has matched.
+    return finalResults
```

Comment:
You know what I just realized? Can't we just skip this condition in total? It was solely the shortcut, and this way if not all terms matched in the title, the file contents would still not be searched, or did I misread my own logic here …?

Reply:
I think it should be possible. If I understood it correctly, when this condition is met, we didn't find anything in the file anyway, so the following post-processing would be skipped.

However, I'd suggest we keep this as an optimization. Because we still have the `sort()` in line 182, and the `for` loop following it. Maybe my understanding of Javascript is lacking here, but these two would still consume some amount of processing power (even though not productive), right? Then having the shortcut, that is just a single comparison, might still serve as an optimization?